packages/sns/lib/utils/snsInitter.ts

```diff
@@ -68,25 +87,41 @@ export async function initSnsSqs(
     }
   }
 
+  const checkPromises: Promise<Either<'not_found', unknown>>[] = []
   // Check for existing resources, using the locators
-  const queuePromise = getQueueAttributes(sqsClient, locatorConfig)
-  const topicPromise = getTopicAttributes(snsClient, locatorConfig.topicArn)
+  const subscriptionTopicArn =
+    locatorConfig.topicArn ?? (await getTopicArnByName(snsClient, locatorConfig.topicName))
+  const topicPromise = getTopicAttributes(snsClient, subscriptionTopicArn)
+  checkPromises.push(topicPromise)
+
+  if (locatorConfig.queueUrl) {
+    const queuePromise = getQueueAttributes(
+      sqsClient,
+      (locatorConfig as SQSQueueLocatorType).queueUrl,
+    )
+    checkPromises.push(queuePromise)
+  }
 
-  const [queueCheckResult, topicCheckResult] = await Promise.all([queuePromise, topicPromise])
+  const [topicCheckResult, queueCheckResult] = await Promise.all(checkPromises)
 
-  if (queueCheckResult.error === 'not_found') {
+  if (queueCheckResult?.error === 'not_found') {
     throw new Error(`Queue with queueUrl ${locatorConfig.queueUrl} does not exist.`)
   }
   if (topicCheckResult.error === 'not_found') {
     throw new Error(`Topic with topicArn ${locatorConfig.topicArn} does not exist.`)
   }
 
-  const splitUrl = locatorConfig.queueUrl.split('/')
-  const queueName = splitUrl[splitUrl.length - 1]
+  let queueName: string
+  if ((locatorConfig as SQSQueueLocatorType).queueUrl) {
+    const splitUrl = (locatorConfig as SQSQueueLocatorType).queueUrl.split('/')
+    queueName = splitUrl[splitUrl.length - 1]
+  } else {
+    queueName = creationConfig!.queue.QueueName!
+  }
 
   return {
     subscriptionArn: locatorConfig.subscriptionArn,
-    topicArn: locatorConfig.topicArn,
+    topicArn: subscriptionTopicArn,
     queueUrl: locatorConfig.queueUrl,
```

Comment:
Why does it mix the configuration of both: `locatorConfig` and `creationConfig`? Both can be provided and we return `queueName` from creation config but queue url comes from locate config.

Reply:
I think we only follow this branch in case we ended up using locate config, but I might be wrong, worth checking